src/CUDA/gpu_vector.cpp

```diff
@@ -24,13 +24,19 @@ namespace gpu
 cuda_memory_manager_type cuda_memory_manager;
 
 void*
-cuda_memory_manager_type::allocate(size_t bytes, std::string name)
+cuda_memory_manager_type::allocate(size_t bytes, std::string name, bool managed)
```

Comment:
CUDA managed memory has many kinds.
Please avoid putting the flag 'managed' and create an interface instead

Please split the allocation tracking from the allocate function.
Create an allocateManaged interface for calling cudaMallocManaged and taking an argument for the cudaMemAttachGlobal or other kinds.

Reply:
I can do that.

Reply:
Done, part of the last round of commits.